src/React/Widgets/AnnotationStoreEditorWidget/index.js

```diff
@@ -41,7 +42,8 @@ export default function annotationStoreEditorWidget(props) {
 
   if (props.annotation && props.annotations[props.annotation.id]) {
     const storedSelectedAnnotation = props.annotations[props.annotation.id];
-    if (storedSelectedAnnotation.generation === props.annotation.generation) {
+    if (storedSelectedAnnotation.generation === props.annotation.generation ||
+        deepEquals(Object.assign({}, storedSelectedAnnotation, { generation: props.annotation.generation }), props.annotation)) {
```

Comment:
I kind of see this as losing some control over the widget.  The generation number allows the developer to choose which properties they care about, rather than force every property to match at all levels.  However, I'd be fine with this if the component could take a prop "forceDeepEquality" which defaults to false.  Or maybe better would be "ignoreGeneration" which would likewise default to false.  But if this prop were true, then your `or` clause could happen.  What do you think?